ivy/functional/ivy/general.py

```diff
@@ -4086,8 +4086,9 @@ def _get_devices_and_dtypes(fn, recurse=False, complement=True):
                 list(fn_unsupported_dnd.values())[0], tuple
             )
 
-        for device, dtypes in fn_unsupported_dnd.items():
-            fn_unsupported_dnd[device] = tuple(_expand_typesets(dtypes))
+        if isinstance(fn_unsupported_dnd, dict):
```

Comment:
Great spot, let's also make the same change on line [`4072`](https://github.com/unifyai/ivy/pull/27125/files#diff-f6556baf0d4607e70527506a4c173c78f8e3452e3b1c10d4d356210c527b41c3R4072) so that it works with the supported dtypes decorators too. Thanks @Sam-Armstrong 😄